### ISSUE TITLE
pylint failing on Python 3.5

### ISSUE THREAD

**Issue by jonparrott** [PyCQA/pylint#1216]
### Steps to reproduce
1. clone https://github.com/GoogleCloudPlatform/google-auth-library-python
2. run `tox -e lint`

### Current behavior

Fails with the following stacktrace:

```
lint runtests: commands[2] | python scripts/run_pylint.py
Traceback (most recent call last):
  File ".tox/lint/lib/python3.5/site-packages/astroid/manager.py", line 172, in file_from_module_name
    value = self._mod_file_cache[(modname, contextfile)]
KeyError: ('google.auth._default', None)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File ".tox/lint/bin/pylint", line 11, in <module>
    sys.exit(run_pylint())
  File ".tox/lint/lib/python3.5/site-packages/pylint/__init__.py", line 13, in run_pylint
    Run(sys.argv[1:])
  File ".tox/lint/lib/python3.5/site-packages/pylint/lint.py", line 1311, in __init__
    linter.generate_reports()
  File "/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/contextlib.py", line 77, in __exit__
    self.gen.throw(type, value, traceback)
  File ".tox/lint/lib/python3.5/site-packages/pylint/lint.py", line 1133, in fix_import_path
    yield
  File ".tox/lint/lib/python3.5/site-packages/pylint/lint.py", line 1310, in __init__
    linter.check(args)
  File ".tox/lint/lib/python3.5/site-packages/pylint/lint.py", line 732, in check
    self._do_check(files_or_modules)
  File ".tox/lint/lib/python3.5/site-packages/pylint/lint.py", line 863, in _do_check
    self.check_astroid_module(ast_node, walker, rawcheckers, tokencheckers)
  File ".tox/lint/lib/python3.5/site-packages/pylint/lint.py", line 944, in check_astroid_module
    walker.walk(ast_node)
  File ".tox/lint/lib/python3.5/site-packages/pylint/utils.py", line 944, in walk
    self.walk(child)
  File ".tox/lint/lib/python3.5/site-packages/pylint/utils.py", line 941, in walk
    cb(astroid)
  File ".tox/lint/lib/python3.5/site-packages/pylint/checkers/variables.py", line 1055, in visit_importfrom
    module = self._check_module_attrs(node, module, name_parts[1:])
  File ".tox/lint/lib/python3.5/site-packages/pylint/checkers/variables.py", line 1125, in _check_module_attrs
    module = next(module.getattr(name)[0].infer())
  File ".tox/lint/lib/python3.5/site-packages/astroid/scoped_nodes.py", line 92, in decorator
    nodes = [n for n in func(*args, **kwargs) if not isinstance(n, cls)]
  File ".tox/lint/lib/python3.5/site-packages/astroid/scoped_nodes.py", line 434, in getattr
    return [self.import_module(name, relative_only=True)]
  File ".tox/lint/lib/python3.5/site-packages/astroid/scoped_nodes.py", line 492, in import_module
    return MANAGER.ast_from_module_name(absmodname)
  File ".tox/lint/lib/python3.5/site-packages/astroid/manager.py", line 116, in ast_from_module_name
    filepath, mp_type = self.file_from_module_name(modname, context_file)
  File ".tox/lint/lib/python3.5/site-packages/astroid/manager.py", line 176, in file_from_module_name
    modname.split('.'), context_file=contextfile)
  File ".tox/lint/lib/python3.5/site-packages/astroid/modutils.py", line 369, in file_info_from_modpath
    return _file_from_modpath(modpath, path, context)
  File ".tox/lint/lib/python3.5/site-packages/astroid/modutils.py", line 582, in _file_from_modpath
    mtype, mp_filename = _module_file(modpath, path)
  File ".tox/lint/lib/python3.5/site-packages/astroid/modutils.py", line 670, in _module_file
    stream, mp_filename, mp_desc = imp.find_module(modname, path)
  File ".tox/lint/lib/python3.5/imp.py", line 270, in find_module
    "not {}".format(type(name)))
RuntimeError: 'list' must be None or a list, not <class 'str'>
Pylint failed on Library with status 1.
```

### Expected behavior

I expected it to lint and yell at me about *my* code. :)

### pylint --version output

```
$ ./.tox/lint/bin/pylint --version
No config file found, using default configuration
pylint 1.6.4,
astroid 1.4.8
Python 3.5.1 (default, Jan 22 2016, 11:49:07)
[GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)]
```

### Some herrings

1. This only occurs on 3.5, it doesn't occur with 3.4 (and we don't test against 3.3).
1. This could be related to https://github.com/pypa/setuptools/issues/885 as we saw the issues arise at the same time, but maybe not. 
1. google-auth is a namespace package, and those always cause all kinds of fun. :)

Please let me know if I can do anything to help reproduce or fix.

**Comment by PCManticore** [comment]
This is interesting, will take a look!

**Comment by jonparrott** [comment]
Thanks!

**Comment by PCManticore** [comment]
This seems to work like a charm on pylint==2.0.0 and astroid==1.5.0. We actually rewrote the import logic, so probably it is not affected by this bug. Indeed, it seems caused by namespace packages, since we pass inadvertently a NamespacePath object to imp.find_module, leading to this error you are noticing.
Can you give it a go with the latest code and let me know if it works for you? I'm closing the issue in the meantime, but feel free to reopen if you can reproduce it.

**Comment by jonparrott** [comment]
@PCManticore is 2.0.0 not yet released?

**Comment by jonparrott** [comment]
@PCManticore seems so - using the latest code from git for `pylint` and `astroid` I get the following:

```
tox --recreate -e lint
GLOB sdist-make: /Users/jonwayne/workspace/google-auth/setup.py
lint recreate: /Users/jonwayne/workspace/google-auth/.tox/lint
lint installdeps: flake8, flake8-import-order, git+git://github.com/PyCQA/astroid.git, git+git://github.com/PyCQA/pylint.git, docutils
lint inst: /Users/jonwayne/workspace/google-auth/.tox/dist/google-auth-0.5.0.zip
lint installed: astroid==1.5.0,docutils==0.13.1,editdistance==0.3.1,flake8==3.2.1,flake8-import-order==0.11,google-auth==0.5.0,isort==4.2.5,lazy-object-proxy==1.2.2,mccabe==0.5.3,pyasn1==0.1.9,pyasn1-modules==0.0.8,pycodestyle==2.2.0,pyflakes==1.3.0,pylint==2.0.0,rsa==3.4.2,six==1.10.0,wrapt==1.10.8
lint runtests: PYTHONHASHSEED='3650788564'
lint runtests: commands[0] | python setup.py check --metadata --restructuredtext --strict
running check
lint runtests: commands[1] | flake8 --import-order-style=google --application-import-names=google,tests,system_tests google tests
lint runtests: commands[2] | python /Users/jonwayne/workspace/google-auth/scripts/run_pylint.py
Traceback (most recent call last):
  File "/Users/jonwayne/workspace/google-auth/.tox/lint/bin/pylint", line 11, in <module>
    load_entry_point('pylint==2.0.0', 'console_scripts', 'pylint')()
  File "/Users/jonwayne/workspace/google-auth/.tox/lint/lib/python3.5/site-packages/pylint/__init__.py", line 13, in run_pylint
    Run(sys.argv[1:])
  File "/Users/jonwayne/workspace/google-auth/.tox/lint/lib/python3.5/site-packages/pylint/lint.py", line 1301, in __init__
    linter.generate_reports()
  File "/Users/jonwayne/homebrew/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/contextlib.py", line 77, in __exit__
    self.gen.throw(type, value, traceback)
  File "/Users/jonwayne/workspace/google-auth/.tox/lint/lib/python3.5/site-packages/pylint/lint.py", line 1124, in fix_import_path
    yield
  File "/Users/jonwayne/workspace/google-auth/.tox/lint/lib/python3.5/site-packages/pylint/lint.py", line 1300, in __init__
    linter.check(args)
  File "/Users/jonwayne/workspace/google-auth/.tox/lint/lib/python3.5/site-packages/pylint/lint.py", line 726, in check
    self._do_check(files_or_modules)
  File "/Users/jonwayne/workspace/google-auth/.tox/lint/lib/python3.5/site-packages/pylint/lint.py", line 855, in _do_check
    self.check_astroid_module(ast_node, walker, rawcheckers, tokencheckers)
  File "/Users/jonwayne/workspace/google-auth/.tox/lint/lib/python3.5/site-packages/pylint/lint.py", line 934, in check_astroid_module
    walker.walk(ast_node)
  File "/Users/jonwayne/workspace/google-auth/.tox/lint/lib/python3.5/site-packages/pylint/utils.py", line 997, in walk
    self.walk(child)
  File "/Users/jonwayne/workspace/google-auth/.tox/lint/lib/python3.5/site-packages/pylint/utils.py", line 994, in walk
    cb(astroid)
  File "/Users/jonwayne/workspace/google-auth/.tox/lint/lib/python3.5/site-packages/pylint/checkers/imports.py", line 410, in visit_importfrom
    self._add_imported_module(node, '%s.%s' % (importedmodnode.name, name))
  File "/Users/jonwayne/workspace/google-auth/.tox/lint/lib/python3.5/site-packages/pylint/checkers/imports.py", line 641, in _add_imported_module
    module_file)
  File "/Users/jonwayne/workspace/google-auth/.tox/lint/lib/python3.5/site-packages/astroid/modutils.py", line 437, in get_module_part
    context_file=context_file)
  File "/Users/jonwayne/workspace/google-auth/.tox/lint/lib/python3.5/site-packages/astroid/modutils.py", line 338, in file_from_modpath
    return file_info_from_modpath(modpath, path, context_file).location
  File "/Users/jonwayne/workspace/google-auth/.tox/lint/lib/python3.5/site-packages/astroid/modutils.py", line 383, in file_info_from_modpath
    return _spec_from_modpath(modpath, path, context)
  File "/Users/jonwayne/workspace/google-auth/.tox/lint/lib/python3.5/site-packages/astroid/modutils.py", line 597, in _spec_from_modpath
    found_spec = spec.find_spec(modpath, [context])
  File "/Users/jonwayne/workspace/google-auth/.tox/lint/lib/python3.5/site-packages/astroid/interpreter/_import/spec.py", line 258, in find_spec
    submodule_path or path)
  File "/Users/jonwayne/workspace/google-auth/.tox/lint/lib/python3.5/site-packages/astroid/interpreter/_import/spec.py", line 220, in _find_spec_with_path
    spec = finder.find_module(modname, module_parts, processed, submodule_path)
  File "/Users/jonwayne/workspace/google-auth/.tox/lint/lib/python3.5/site-packages/astroid/interpreter/_import/spec.py", line 87, in find_module
    stream, mp_filename, mp_desc = imp.find_module(modname, submodule_path)
  File "/Users/jonwayne/workspace/google-auth/.tox/lint/lib/python3.5/imp.py", line 270, in find_module
    "not {}".format(type(name)))
RuntimeError: 'list' must be None or a list, not <class 'str'>
Pylint failed on Library with status 1.
ERROR: InvocationError: '/Users/jonwayne/workspace/google-auth/.tox/lint/bin/python /Users/jonwayne/workspace/google-auth/scripts/run_pylint.py'
```

**Comment by PCManticore** [comment]
@jonparrott This should be fixed in astroid==1.4.9 and in the master branch.

**Comment by jonparrott** [comment]
@PCManticore seems good now, thanks!
